src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -210,12 +210,12 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo)
             }
         case CONDITION_NEAR_CREATURE:
             {
-                condMeets = !!GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, !ConditionValue3);
+                condMeets = static_cast<bool>(GetClosestCreatureWithEntry(object, ConditionValue1, (float) ConditionValue2,!ConditionValue3) != nullptr);
                 break;
             }
         case CONDITION_NEAR_GAMEOBJECT:
             {
-                condMeets = !!GetClosestGameObjectWithEntry(object, ConditionValue1, (float)ConditionValue2);
+                condMeets = static_cast<bool>(GetClosestGameObjectWithEntry(object, ConditionValue1, (float)ConditionValue2));
```

Comment:
why above you added the `!= nullptr` and here no?

Reply:
fixed